src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -418,15 +431,27 @@ private async Task<JObject> GetLocalSettingsJsonAsJObjectAsync()
         public override async Task RunAsync()
         {
             await PreRunConditions();
+            var isVerbose = VerboseLogging.HasValue && VerboseLogging.Value;
 
             var isCurrentProcessNet6Build = RuntimeInformation.FrameworkDescription.Contains(Net6FrameworkDescriptionPrefix);
-            if (isCurrentProcessNet6Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            // If --runtime param is set, handle runtime param logic; otherwise we infer the host to launch on startup
+            if (SetHostRuntime != null)
             {
-                await StartInProc8AsChildProcessAsync();
-                return;
+                var shouldReturn = await ShouldReturnAfterHandlingRuntimeAsync(isCurrentProcessNet6Build, isVerbose);
+                if (shouldReturn)
+                {
+                    return;
```

Comment:
What happens if we return here? What is the behaviour?

Reply:
updated naming to make it more accurate!